tests/functional/src/main/java/org/mule/functional/api/classloading/isolation/AetherClassPathClassifier.java

```diff
@@ -0,0 +1,692 @@
+                                                extensionsInfrastructure));
+        }
+      }
+    }
+
+    File generatedResourcesDirectory = new File(baseResourcesFolder, separator + "META-INF");
+    extensionsInfrastructure.generateDslResources(generatedResourcesDirectory);
+    return pluginUrlClassifications;
+  }
+
+  private boolean isExtensionPlugin(Artifact artifact) {
+    return artifact.getExtension().equals(MULE_EXTENSION);
+  }
+
+  /**
+   * Classifies a plugin {@link Artifact}. {@value org.eclipse.aether.util.artifact.JavaScopes#COMPILE} dependencies will be
+   * resolved for building the {@link URL}'s for the class loader. For {@link Extension} annotated classes it will also generate
+   * its metadata.
+   *
+   * @param pluginArtifact {@link Artifact} that represents the plugin to be classified
+   * @param context {@link ClassPathClassifierContext} with settings for the classification process
+   * @param baseResourcesFolder base {@link File} folder for extensions metadata
+   * @param extensionsInfrastructure {@link ExtensionsTestInfrastructureDiscoverer} to generate metadata
+   * @return {@link PluginUrlClassification} for the plugin
+   */
+  private PluginUrlClassification buildPluginUrlClassification(Artifact pluginArtifact, ClassPathClassifierContext context,
+                                                               File baseResourcesFolder,
+                                                               ExtensionsTestInfrastructureDiscoverer extensionsInfrastructure) {
+    List<Dependency> managedDependencies = dependencyResolver.readArtifactDescriptor(pluginArtifact).getManagedDependencies();
+
+    List<URL> urls = toUrl(dependencyResolver
+        .resolveDependencies(
+                             new Dependency(pluginArtifact,
+                                            COMPILE),
+                             Collections.<Dependency>emptyList(),
+                             managedDependencies,
+                             orFilter(classpathFilter(COMPILE),
+                                      new PatternExclusionsDependencyFilter(context.getExcludedArtifacts()))));
+
+    if (context.isExtensionMetadataGenerationEnabled()) {
+      URL generatedTestResources =
+          buildExtensionPluginMetadata(baseResourcesFolder, extensionsInfrastructure, pluginArtifact, urls);
+
+      if (generatedTestResources != null) {
+        List<URL> appendedTestResources = newArrayList(generatedTestResources);
+        appendedTestResources.addAll(urls);
+        urls = appendedTestResources;
+      }
+    }
+    // TODO (gfernandes): How could I check if exported classes belong to this plugin?
+    return new PluginUrlClassification(toId(pluginArtifact), urls, newArrayList(context.getExportPluginClasses()));
+  }
+
+  /**
+   * @return an {@link ExtensionManagerAdapter} that would be used to register the extensions, later it would be discarded.
```

Comment:
Is not clear to me why it is discarded and why it is needed on the first place.  Can you clarify that?